tests/Microsoft.Bot.Builder.TestBot/TestBot.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Microsoft.Recognizers.Text;
+
+namespace Microsoft.Bot.Builder.TestBot
+{
+    public class TestBot : IBot
+    {
+        private TestBotAccessors _accessors;
+
+        public TestBot(TestBotAccessors accessors)
+        {
+            _accessors = accessors;
+        }
+
+        public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (turnContext.Activity.Type == ActivityTypes.Message)
+            {
+                // create the DialogSet 
+                await _accessors.ConversationDialogState.GetAsync(turnContext, () => new DialogState());
+                var dialogs = new DialogSet(_accessors.ConversationDialogState);
```

Comment:
Hmm, is this really representative of the way the dialog API is meant to be used? @Stevenic was telling me that it's intended to be created up front and used as a singleton, not instantiated on every turn. 

Instantiating the entire hierarchy of any sufficiently complex `DialogSet` on every turn is going to be a problemmatic for performance, especially GC.

Reply:
@drub0y, John and I talked offline and we changed the code to fall more in line with what you mention.  John has PR #863 out which updates the code.